File: IGListTest/IGListTest/Views/JournalEntryDateCell.m
//
//  JournalEntryDateCell.m
//  IGListTest
//
//  Created by liuruixuan on 2023/4/16.
//

#import "JournalEntryDateCell.h"
#import "Utils.h"

@interface JournalEntryDateCell ()

@end

@implementation JournalEntryDateCell

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        self.contentView.backgroundColor = ssRGBHex(0x0c1f3f);
        [self.contentView addSubview:self.label];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
    self.label.frame = UIEdgeInsetsInsetRect(self.bounds, UIEdgeInsetsMake(0, CommonInsets.left, 0, CommonInsets.right));
}

- (UILabel *)label {
    if (!_label) {
        _label = [[UILabel alloc] init];
        _label.font = AppFont(14);
        _label.textColor = ssRGBHex(0x42c84b);
        _label.backgroundColor = [UIColor clearColor];
    }
    return _label;
}

@end

File: Marslink_Swift5/Marslink_Swift5/Views/JournalEntryDateCell.swift
//
//  JournalEntryDateCell.swift
//  Marslink_Swift5
//
//  Created by liuruixuan on 2023/4/17.
//

import UIKit

class JournalEntryDateCell: UICollectionViewCell {
    let label: UILabel = {
        let label = UILabel()
        label.backgroundColor = UIColor.clear
        label.font = AppFont(size: 14)
        label.textColor = UIColor(hex6: 0x42C84B)
        return label
    }()

    override init(frame: CGRect) {
        super.init(frame: frame)
        contentView.backgroundColor = UIColor(hex6: 0x0C1F3F)
        contentView.addSubview(label)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func layoutSubviews() {
        super.layoutSubviews()
        let padding = CommonInsets
        label.frame = bounds.inset(by: UIEdgeInsets(top: 0, left: padding.left, bottom: 0, right: padding.right))
    }
}
